### PR TITLE
Check provider versions on policy runs

### PR DESCRIPTION
**Pull request by irmatov**

### CODE REVIEW

**Review by fdistorted — APPROVED** [review]
looks good

File: pkg/client/client.go [hunk 1]
```diff
@@ -639,3 +653,21 @@ func normalizeResources(requested []string, all map[string]*schema.Table) ([]str
...
+			return nil, err
+		}
+		v, err := version.NewVersion(s)
+		if err != nil {
+			return nil, err
+		}
+		ver[p.Name] = v
+	}
+	return ver, nil
+}
```
[the remark this answers is not in the document]
Comment (roneli) [reply pkg/client/client.go:788]: I think also drift code added something of the same, maybe combine both functions in a single file?

**Review by roneli — CHANGES_REQUESTED** [review]
Looks good, some minor changes needed

File: pkg/client/client.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (irmatov) [reply pkg/client/client.go:788]: > maybe move version related functions in a file called version.go under client package ?

It is only one function, collectProviderVersions. I don't think it'll improve anything with such a small amount of code. Also there is already a version.go with an unrelated variable.

> I think also drift code added something of the same, maybe combine both functions in a single file?

I can't see anything that can be shared/ combined.

File: pkg/policy/execute.go [hunk 2]
```diff
@@ -158,50 +164,63 @@ func (e *Executor) ExecuteQuery(ctx context.Context, q *config.Query) (*QueryRes
...
 		values, err := data.Values()
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("%s: %w", q.Name, err)
 		}
 		result.Data = append(result.Data, values)
 	}
 	if data.Err() != nil {
-		return nil, data.Err()
+		return nil, fmt.Errorf("%s: %w", q.Name, data.Err())
```
[the remark this answers is not in the document]
Comment (irmatov) [reply pkg/policy/execute.go:172]: What info exactly?

policy/query: syntax error
policy/query: can't convert X to Y
policy/query: connection reset

vs

policy/query: error executing query: syntax error
policy/query: error parsing query result: can't convert X to Y
policy/query: final error: connection reset

doesn't improve much except adding more noise. (all three cases above are easily distinguished by the error itself).

**Review by roneli — APPROVED** [review]
Will want also @zagronitay to have a look since he is touching the same code base

**Pull request merged**